mlir/include/mlir/Dialect/EmitC/IR/EmitCTypes.td

```diff
@@ -130,4 +130,16 @@ def EmitC_PointerType : EmitC_Type<"Pointer", "ptr"> {
   let assemblyFormat = "`<` qualified($pointee) `>`";
 }
 
+class EmitC_SizeType<string name, string typeMnemonic, list<Trait> traits = []> 
```

Comment:
I don't really see the benefit of this declaration; I think we can remove it and make the two below directly inherit from EmitC_Type?

Reply:
I'd like to be able to use it actually so we can check membership of both types at once.

Reply:
But can you?

Reply:
I haven't been able to... but it looks a bit odd to me that such type hierarchies only hold within TableGen and you can't use them outside. Again, there must be a good reason why.